packages/Icon/src/createIconComponent.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default glyphs => {
```

Comment:
nit: IMO this will be easier to read + find if it was made a function declaration with a name. i.e. `export default function createIconComponent(glyphs) { .. }`. Such that if someone is grepping the code base then they'd find this function. It also highlights that this is _the thing_ that is being exported. As an anonymous function it gets kinda lost in the other noise of the function.

Reply:
Thought: what if we allowed this argument to also accept any valid React Node, such that someone could do `createIconComponent(() => <svg />)` and get back an `Icon` which has a fixed value for its `glyph` prop? May not be necessary now so no need to bloat, but wanted to raise it as a question.

Reply:
Funny you mention that, I was actually talking with a friend of mine about that pattern over the weekend. Really wasn't something I had thought about much prior, but I love it as a pattern going forward!

Reply:
Done